libs/http-rpc/http-rpc/src/main/kotlin/net/corda/httprpc/JsonObject.kt

```diff
@@ -0,0 +1,32 @@
+package net.corda.httprpc
+
+/**
+ * Marker interface which can be used in HTTP request / response objects to indicate a field is a JSON object. As a result, such fields
+ * accept the object as a JSON object or a string version of the JSON with escaped quotation marks.
+ *
+ * For example, if the request object is defined as follows:
+ *
+ * ```
+ * data class RequestWithJsonObject(val id: String, val obj: JsonObject)
+ * ```
+ *
+ * Then the following payloads will be acceptable:
+ *
+ * A - "obj" is a JSON object.
+ * ```
+ * {
+ *   "id": "real-object",
+ *   "obj": {"message": "Hey Mars", "planetaryOnly":"true", "target":"C=GB, L=FOURTH, O=MARS, OU=PLANET"}
+ * }
+ * ```
+ * B - "obj" is a String containing JSON with escaped quotation marks.
+ * ```
+ * {
+ *   "id": "string-escaped",
+ *   "obj": "{\"message\":\"Hey Mars\", \"planetaryOnly\":\"true\", \"target\":\"C=GB, L=FOURTH, O=MARS, OU=PLANET\"}"
+ * }
+ * ```
+ *
+ * Internally, implementations of [RpcOps] endpoints should use `toString()` on the [JsonObject] to get access to the object's JSON.
+ */
+interface JsonObject
```

Comment:
I wonder if it would be neater to change this interface as follows:
```suggestion
interface JsonObject {
    val escapedJson: String
}
```
then we can use on `escapedJson` rather than rely on built-in `toString()` function.

Reply:
I didn't go with that approach because openApi generates a "JsonObject" with a property "escapedJson" and the purpose here is to allow real JSON rather than escaped JSON inside an object. The fact it escapes the Json in the deserializer is just internal. I don't think we need to expose this on the public API. 

The example payload in swagger if we did that:
```
{
  "clientRequestId": "string",
  "flowClassName": "string",
  "requestData": {
    "escapedJson": "string"
  }
}
```
`escapedJson` is a non-nullable string field and people would start using it like this:
```
{ 
    ...
    "requestData": {
        "escapedJson": "{\"command\":\"echo\", \"data\":{\"echo_value\": \"helloconal\"}}"
    }
}
```
When in reality we want them to use it like this:
```
{ 
    ...
    "requestData": "{\"command\":\"echo\", \"data\":{\"echo_value\": \"helloconal\"}}"
}
```
or
```
{ 
    ...
    "requestData": {
        "command":"echo", 
        "data":{
            "echo_value": "helloconal"
        }
    }
}
```
On the other hand, the current approach generates example request body with just empty object {} and this might be a bit confusing. Maybe we can improve this.
```
{
  "clientRequestId": "string",
  "flowClassName": "string",
  "requestData": {}
}
```

![image](https://user-images.githubusercontent.com/68279385/185452444-83fec4b4-e82c-4e5b-bff0-083b9647ac35.png)

Reply:
Yeah, I was wondering what would happen to the example JSON in this case.

Maybe having the ability to specify example request bodies might be needed for these APIs? Not sure how else they can generate anything meaningful.

Reply:
I think there are two related aspects here:
1. The code in Kotlin that is easy to reason about;
2. Correct and meaningful representation of data structures in OpenAPI.

As for first point I think changing interface to:
```
interface JsonObject {
    val escapedJson: String
}
```
will make code look neater rather than we currently relying on `toString()` function. I totally appreciate that it has a negative impact on the OpenAPI we generate.

Which brings me to the second point, where we can add a custom `net.corda.httprpc.server.impl.apigen.processing.openapi.schema.builders.SchemaBuilder` for `JsonObject` where we will be able to control what gets written to OpenAPI. We can even provide a reasonable example here, I think.

Reply:
Yes that sounds reasonable, I'll apply those changes